### PR TITLE
fix(replica_rebuild): check replica state before checking wait duration

### PR DESCRIPTION
**Pull request by abhilashshetty04**

### CODE REVIEW

**Comment by abhilashshetty04** [comment]
bors try

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs [hunk 1]
```diff
@@ -164,18 +164,19 @@ async fn handle_child_rebuild(
 ) -> Result<(), SvcError> {
     let wait_duration = RuleSet::faulted_child_wait(nexus, context.registry());
     let is_elapsed = wait_duration_elapsed(child.faulted_at, wait_duration, &child.uri);
-    if child.state_reason == ChildStateReason::RebuildFailed || is_elapsed {
-        info!(%child.uri, "Start full rebuild for child, elapsed: {}, child state reason: {:?}", is_elapsed.to_string(), child.state_reason);
-        faulted_children_remover(nexus_spec, child, context).await?
-    } else if get_child_replica(nexus_spec.as_ref(), child, context)
-        .await
-        .is_ok()
+    if (child.state_reason.clone() != ChildStateReason::RebuildFailed)
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs]: Can we move RebuildFailed to a separate condition so we don't have to keep checking?
```suggestion
    if (child.state_reason == ChildStateReason::RebuildFailed) { return fault... }
```

File: control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs [hunk 2]
```diff
@@ -164,18 +164,19 @@ async fn handle_child_rebuild(
...
+        & get_child_replica(nexus_spec.as_ref(), child, context)
```
Comment (dsharma-dc) [inline control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs]: logical AND `&&`

File: control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs (hunk 1 above)
Comment (dsharma-dc) [reply control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs]: or take it in a variable in the beginning like is_elapsed:
`let is_rebuild_failed = (child.state_reason == ChildStateReason::RebuildFailed);`

**Review by dsharma-dc — APPROVED** [review]

**Comment by abhilashshetty04** [comment]
bors try

**Comment by abhilashshetty04** [comment]
bors merge

**Pull request merged**